### ISSUE TITLE
Little delays and Random lights when nothing is playing as well.

### ISSUE THREAD

**Issue by ramkithepower** [scottlawsonbc/audio-reactive-led-strip#68]
I have got this up and running and I can really see the led strip reacting to the music. But there are some lights showing up even when there is no song or sound playing in the PC. I will put up a youtube link as soon as possible. 

Is there any reason why it may be slow. I am running from a powerful PC with the GUI running in the PC as well.

**Comment by nickmc1171** [comment]
Is the delay happening when processing or controlling the lights? As for the random lights, you can try upping the MIN_VOLUME_THRESHOLD in config.

**Comment by ramkithepower** [comment]
By disabling the display of FPS am able to get a decent output. Not sure if the print is consuming some bit of time which adds on to the delay

**Comment by nickmc1171** [comment]
Interesting. I've never thought of printing to console as a resource intensive task, but I guess it's significant enough to cause a delay. If fps is important to you, you could try lowering the frequency of its output. Just out of curiosity, what OS and terminal are you using, and what's your screen size/resolution?

**Comment by ramkithepower** [comment]
I wouldn't have thought so as well. Mine is a Windows 10 pc running of i7
processor and 12 gigs of RAM.

**Comment by nickmc1171** [comment]
Yeah, that should be more than enough for this. Well, I hope you get it all sorted. I'm not home right now, but I can run some tests or something this weekend if it will help.

**Comment by ramkithepower** [comment]
Yeah that would be great. I will set the mic threshold and see as well..

**Comment by nickmc1171** [comment]
Ok cool. I set mine to 1e-4, but I'm sure it varies by pc and input device.

**Comment by ramkithepower** [comment]
Tried that as well.. This is what happens....

https://www.youtube.com/edit?o=U&video_id=-rXyAfl9Co8

**Comment by nickmc1171** [comment]
That's exactly what mine was doing before i upped the threshold. I'm also not using a microphone for audio in though. How familiar are you with python/coding? Can you step through the code and see what it's reporting as volume when there is no audio in?

**Comment by ramkithepower** [comment]
I am very new to python, in fact only started looking into python because of this project. But am a good code builder, so if you let me know what i should do i may be able to do it.

With respect to your question, I tried putting the following print statement after the volume deriving formula
vol = np.max(np.abs(y_data))
print('Current Volume:', vol)

and the output was changing between 0.000183105 and 0.000213623
![image](https://user-images.githubusercontent.com/30769165/29701282-6be22302-8988-11e7-8ce6-e79d58b3d15e.png)

**Comment by ramkithepower** [comment]
I will do another test and let you know in 12 hours. I just ran a sample again and can see the output even when the volume is below threshold

**Comment by nickmc1171** [comment]
Ok. We should probably move this off of github so we don't clutter up the forum. If you want to give me your email I can start a Slack channel and invite you. Or you can start something and invite me if you don't want to post your email. Mine is [my github username]@hotmail.com

**Comment by ramkithepower** [comment]
Sure. Just mailed you...

**Comment by tettar** [comment]
Hello, i have the same issue decribe in the youtube video by  ramkithepower 👍 
Same Os, same issue ...
Did you find the root cause ?

**Comment by ramkithepower** [comment]
Yes it was the microphone volume

**Comment by tettar** [comment]
thx  ramkithepower i set  up MIN_VOLUME_THRESHOL in config.py

MIN_VOLUME_THRESHOLD = 1e-2

I thought that the problem resulted from the background noise and that this could be filter in another way !!!

But like this, all is ok, 

thx :)

**Comment by SaschaBuettner** [comment]
Hi, I'll drop in here as the problem seems similar. I use a Raspberry Pi 2 B+ with a 120 LED WS2182B strip on a recent raspbian lite and a slim USB microphone which identifies as C-media audio device. While all runs very well, I have the "unexpected output when silence problem" too. While playing with threshold I do not get an acceptabe result. I need to put in a very high value like 0.08 to get a black strip in silence. Unfortunately this results in heavy flashing during more silent passages of music. What I am looking for is some kind of control over the auto gain process. 

Is there any way to limit the maximum added gain? Let's say the loudness drops below -42dB then the alsa system should no longer try to amplify the input signal until it reaches a higer value again? Some kind of gating function?

So I could keep the Threshold value low enough to get a smooth operation?
